.env.development

```diff
@@ -12,3 +12,5 @@ OKP4_TELEGRAM_URL='https://t.me/okp4network'
 OKP4_TWITTER_URL='https://twitter.com/OKP4_Protocol'
 OKP4_WEBSITE_URL='https://okp4.network/'
 DATA_PATH='./data'
+API_URI='http://localhost:3000/api'
+NEXT_PUBLIC_SERVER_URI='http://localhost:3000'
```

Comment:
I can't figure out why you need to prefix by `NEXT_PUBLIC`. Not correct to me. If you intend to use it to retrieve the configuration (`/api/config`), I suggest you use a relative URL, with respect to the current domain. As it is, this will cause deployment issues.

Reply:
I use `NEXT_PUBLIC` (for `NEXT_PUBLIC_SERVER_URI` env variable) because I've created an util function to get the config. The utils file is not on server side (it is on client side) so we need the prefix `NEXT_PUBLIC` to get the env variable. Here this env variable is the path to the next api.

Because this function (`getConfig()`) can be called inside `getStaticProps()`, we must use the full path to the api folder (port included). This is why I decided to use en .env variable too.

Reply:
The problem with `NEXT_PUBLIC` is following:
>The value will be inlined into JavaScript sent to the browser because of the NEXT_PUBLIC_ prefix. This inlining occurs at build time, so your various NEXT_PUBLIC_ envs need to be set when the project is built.

The value is inlined at build time, so there's no way to change it afterwards on deployment. For this specific value, just consider a relative path. It's fine.

Reply:
Here is why I decided to do it this way: https://github.com/vercel/next.js/discussions/12763 .

If we use a relative path, it won't work if we run the project on another port. Maybe we could declare this variable inside next config. This way allow us to edit it on runtime. What do you think about it ?

Reply:
There's no other port than 443 from outside in prod. If we're talking about the config it's fine to "host" it in the same place than any other resources.

From what I understand, with the current implementation, it's not possible to change the value of the variable after the build. And that's problematic.

Reply:
If we add this variable inside the `next.config.mjs` file in `publicRuntimeConfig` object, will it be ok for you ?

Reply:
@RaphaelEscrig Please use a relative path to fetch the config and remove the `NEXT_PUBLIC_SERVER_URI `.

The config endpoint cannot be provided through another host and port so we don't need to configure it through env.

And by the way, it cannot work as is because once built the variable cannot be changed, so if we deploy it on `https://whatever.host` the config will still be fetch on `localhost:3000`..